thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/datasource/pinot/PinotControllerResponseCacheLoader.java

```diff
@@ -39,6 +40,8 @@
 
   private static final String BROKER_PREFIX = "Broker_";
 
+  private final AtomicInteger activeConnections = new AtomicInteger();
```

Comment:
Just use an int is sufficient because you are using it in a synchronized block; JVM will make sure every thread reads the latest update.

Reply:
we're synchronizing on one of 25 connections. the int is not protected.